specs/deposit_gas_metering.md

```diff
@@ -100,14 +100,15 @@ if curr_num != block.number {
     curr_number := block.number
     curr_bought_gas := 0
     pack_and_store(curr_basefee, curr_number, curr_bought_gas)
-
 }
 
 curr_bought_gas += required_gas
 require(curr_bought_gas <= GUARANTEED_GAS_LIMIT)
 require(curr_bought_gas <= SANITY_GAS_LIMIT)
 gas_cost = requested_gas * curr_basefee
 
+store(curr_bought_gas) # May have to be a pack and store as well.
```

Comment:
if you don't return in the above if statement, then the store and pack thing runs twice. Just move that line to the end and it would work. We don't want to run a single `store` here, because we're packing the data.

Reply:
Yea, but what about the revert?
TBH I don't really consider the pseudocode that important - particularly with the details of how to deal with storing the data.
The basefee calculations are much more important to specify exactly (including how to handle integer divisions)

Reply:
If we revert the deposit then we don't consume actual gas anyway. The effect (storing the update, and the log) should be at the very end of the logic.

Reply:
Also, I liked the python style of the old pseudocode I wrote better. It makes it easier to compare against the EIP-1559 spec: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md

Reply:
That's a difference in taste - I think the python specs are not actually good as specs.

I'm also looking for some to explain why if `parent_gas_used > parent_gas_target` the basefee_delta cannot be 0 while on the flipside of that condition, it can be.
Another fun fact is that go-ethereum clamps the new basefee at zero (i.e. won't let it go below that, but I can't find a reference to that in the spec).

Reply:
I think in the original pseudocode I tried to not have the 0 case at all by adding an if check to both arms. But yes, definitely don't want to change the basefee if the gas matches target exactly

Reply:
Different thing. Your code is exactly the same structure as 1559 w/ renames

```
if used == target:
    basefee_delta = 0
elif used > target:
    basefee_delta = .....
    basefee_delta = max(basefee_delta, 1) # Only in this arm
elif used < target:
   basefee_delta = ....
```

Reply:
Yep, because we want EIP-1559. We should link to the EIP in our spec. And I like less verbose names

Reply:
I have a preference for python style code in specs too, it feels like its less characters and easier to read

Reply:
What is the sanity limit used for here? Its read from the state but unused?